SoomlaiOSStore/StoreEventHandling.m

```diff
@@ -80,12 +80,12 @@ + (void)postGoodUnEquipped:(NSString*)goodItemId{
     [[NSNotificationCenter defaultCenter] postNotificationName:EVENT_GOOD_UNEQUIPPED object:self userInfo:userInfo];
 }
 
-+ (void)postGoodUpgrade:(NSString*)goodItemId withGoodUpgrade:(NSString*)goodUpgradeItemId{
-    NSDictionary *userInfo = [NSDictionary dictionaryWithObjectsAndKeys:
-                              goodItemId, DICT_ELEMENT_GOOD,
-                              goodUpgradeItemId, DICT_ELEMENT_UpgradeVG,
-                              nil];
-    [[NSNotificationCenter defaultCenter] postNotificationName:EVENT_GOOD_UPGRADE object:self userInfo:userInfo];
++ (void)postGoodUpgrade:(NSString*)goodItemId withGoodUpgrade:(NSString*)goodUpgradeItemId {
+    NSMutableDictionary *mutableUserInfo = [NSMutableDictionary dictionaryWithDictionary: @{DICT_ELEMENT_GOOD: goodItemId}];
+    if (goodUpgradeItemId) {
+        mutableUserInfo[DICT_ELEMENT_UpgradeVG] = goodUpgradeItemId;
+    }
+    [[NSNotificationCenter defaultCenter] postNotificationName:EVENT_GOOD_UPGRADE object:self userInfo:mutableUserInfo.copy];
```

Comment:
@EugeneButusov, why does it happen `goodUpgradeItemId` is `nil`. Is not it a configuration issue that we need to catch on an earlier stage?